hudi-common/src/main/java/org/apache/hudi/common/fs/HoodieWrapperFileSystem.java

```diff
@@ -192,27 +261,74 @@ public FSDataOutputStream create(Path f, FsPermission permission, boolean overwr
     return executeFuncWithTimeMetrics(MetricName.create.name(), f, () -> {
       final Path translatedPath = convertToDefaultPath(f);
       return wrapOutputStream(f,
-          fileSystem.create(translatedPath, permission, overwrite, bufferSize, replication, blockSize, progress));
+          fileSystem.create(translatedPath, permission, overwrite, bufferSize, replication, blockSize, progress), bufferSize);
     });
   }
 
-  private FSDataOutputStream wrapOutputStream(final Path path, FSDataOutputStream fsDataOutputStream)
+
+  /**
+   * The stream hierarchy after wrapping will be as follows.
+   *
+   *  FSDataOuputStream (returned)
+   *      BufferedOutputStream (required for output buffering)
+   *          TimedSizeAwareOutputStream  (required for tracking metrics, timings and the number of bytes written)
+   *               fs.open()   (Original stream returned from underlying FileSystem)
+   */
+  private FSDataOutputStream wrapOutputStream(final Path path, FSDataOutputStream fsDataOutputStream, int bufferSize)
       throws IOException {
-    if (fsDataOutputStream instanceof SizeAwareFSDataOutputStream) {
-      return fsDataOutputStream;
+    TimedSizeAwareOutputStream tos = new TimedSizeAwareOutputStream(path, fsDataOutputStream, consistencyGuard,
+        () -> openStreams.remove(path.getName()));
+
+    if (ioBufferingEnabled) {
+      int minBufferSize = (HoodieFileFormat.isBaseFile(path) || FSUtils.isLogFile(path)) ? minDataFileIOBufferSizeBytes
+          : minFileIOBufferSizeBytes;
```

Comment:
Can we just add a method named `FsUtil.isDataFile(Path)` to enclose
```java
HoodieFileFormat.isBaseFile(path) || FSUtils.isLogFile(path)
```

Reply:
Done